### PR TITLE
assure backed candidates are always sorted

### PR DESCRIPTION
**Pull request by drahnr**
At a later point in time, we require the backed candidates are sorted. This adds an extra assurance CC @Lldenaurois 

skip check-dependent-cumulus

### CODE REVIEW

**Comment by drahnr** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4419** 
    * **PR #4421** 
      * **PR #4423** 
        * **PR #4422**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh) and will continue to be automatically updated while this PR remains open.

File: runtime/parachains/src/paras_inherent/mod.rs [hunk 1]
```diff
@@ -900,20 +902,35 @@ fn sanitize_backed_candidates<
...
+			scheduled_paras_to_core_idx.get(&desc.para_id).is_some()
+	});
+
+	// Sort the `Vec` last, once there is a guarantee that these
+	// `BackedCandidates` references the expected relay chain parent,
+	// but more importantly are scheduled for a free core.
+	// This both avoids extra work for obviously invalid candidates,
+	// but also allows this to be done in place.
+	backed_candidates.sort_by(|x, y| {
+		// Never panics, since we filtered all panic arguments out in the previous `fn retain`.
```
Comment (pepyakin) [inline runtime/parachains/src/paras_inherent/mod.rs:931]: You mean the call above?

**Review by pepyakin — APPROVED** [review]

File: runtime/parachains/src/paras_inherent/mod.rs (hunk 1 above)
Comment (drahnr) [reply runtime/parachains/src/paras_inherent/mod.rs:931]: Exactly.

**Comment by drahnr** [comment]
skip check-dependent-cumulus

**Review by Lldenaurois — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

**Comment by drahnr** [comment]
bot merge

**Comment by drahnr** [comment]
bot merge

**Comment by ordian** [comment]
bot merge

**Comment by drahnr** [comment]
Removed it from graphite stacks, rebased and force pushed.

**Comment by drahnr** [comment]
bot merge

**Pull request merged**